src/test/scala/is/hail/rvd/OrderedRVDPartitionerSuite.scala

```diff
@@ -130,4 +130,31 @@ class OrderedRVDPartitionerSuite extends TestNGSuite {
         Interval(Row(11, 0, 2), Row(11, 0, 20), false, false))
     )
   }
+
+  @Test def testIntersect() {
+    val kType = TStruct(("key", TInt32()))
+    val left =
+      new OrderedRVDPartitioner(kType,
+        Array(
+          Interval(Row(1), Row(10), true, false),
```

Comment:
I'm not totally convinced by the logic of intersect, and this test doesn't test situations where endpoints of left and right partitioner coincide. Could you extend it to do so?